src/gc/gc.cpp

```diff
@@ -15535,6 +15542,27 @@ void gc_heap::gc1()
 
     descr_generations (FALSE);
 
+#ifdef GCTRIMCOMMIT
+    if (g_pConfig->GetGCTrimCommit())
+    {
+        decommit_heap_segment_pages(ephemeral_heap_segment, 0);
+
+        heap_segment* seg1 = generation_start_segment (generation_of (max_generation));
+        while (seg1 != ephemeral_heap_segment)
+        {
+            decommit_heap_segment_pages(seg1, 0);
+            seg1 = heap_segment_next (seg1);
+        }
+
+        heap_segment* seg2 = generation_allocation_segment (generation_of (max_generation + 1));
```

Comment:
It seems to me that decommitting the segment that the LOH is allocating upon could be performance trouble - we're just going to have to allocate a new one the next LOH allocation. Then again, I suppose it would be reasonable to expect performance tradeoffs when running with GCTrimCommit.